core/trino-main/src/main/java/io/trino/execution/StageStateMachine.java

```diff
@@ -638,6 +643,36 @@ public StageInfo getStageInfo(Supplier<Iterable<TaskInfo>> taskInfosSupplier)
                 failureInfo);
     }
 
+    private static List<OperatorStats> combineTaskOperatorSummaries(List<TaskInfo> taskInfos, int maxTaskOperatorSummaries)
+    {
+        // Group each unique pipelineId + operatorId combination into lists
+        Long2ObjectOpenHashMap<ArrayList<OperatorStats>> pipelineAndOperatorToStats = new Long2ObjectOpenHashMap<>(maxTaskOperatorSummaries);
+        // Expect to have one operator stats entry for each taskInfo
+        int taskInfoCount = taskInfos.size();
+        LongFunction<ArrayList<OperatorStats>> statsListCreator = key -> new ArrayList<>(taskInfoCount);
```

Comment:
Inlining the lambda into the loop body below will result in a new lambda instance creation per loop iteration since it captures the `taskInfoCount` value. It's a very minor thing though, so I'm happy to inline it anyways if you feel strongly about it.

Reply:
Oh, you are right, missed that. Keep it as is then.